src/python/WMCore/MicroService/Unified/MSOutput.py

```diff
@@ -284,41 +284,56 @@ def makeSubscriptions(self, workflow):
             #    but an iterator of length 'stride' and then we should be doing:
             #    for workflow in workflows:
             if isinstance(workflow, MSOutputTemplate):
-                if workflow['isRelVal']:
-                    pass
-                else:
-                    try:
+                ddmReqList = []
+                try:
+                    if workflow['isRelVal']:
+                        for dMap in workflow['destinationOutputMap']:
+                            ddmRequest = DDMReqTemplate('copy',
+                                                        item=dMap['datasets'],
+                                                        n=workflow['numberOfCopies'],
+                                                        site=dMap['destination'],
+                                                        group='RelVal')
+                            ddmReqList.append(ddmRequest)
+                    else:
                         ddmRequest = DDMReqTemplate('copy',
                                                     item=workflow['OutputDatasets'],
                                                     n=workflow['numberOfCopies'],
                                                     site=workflow['destination'])
-                    except Exception as ex:
-                        msg = "Could not create DDMReq for Workflow: {}".format(workflow['RequestName'])
-                        msg += "Error: {}".format(ex)
-                        self.logger.exception(msg)
-                        return workflow
-
-                    try:
-                        # In the message bellow we may want to put the list of datasets too
-                        msg = "Making transfer subscriptions for {}".format(workflow['RequestName'])
-                        self.logger.info(msg)
-                        ddmResult = self.ddm.makeRequest(ddmRequest)
-                    except Exception as ex:
-                        msg = "Could not make transfer subscription for Workflow: {}".format(workflow['RequestName'])
-                        msg += "Error: {}".format(ex)
-                        self.logger.exception(msg)
-                        return workflow
+                        ddmReqList.append(ddmRequest)
+                except Exception as ex:
+                    msg = "Could not create DDMReq for Workflow: {}".format(workflow['RequestName'])
+                    msg += "Error: {}".format(ex)
+                    self.logger.exception(msg)
+                    return workflow
 
+                try:
+                    # In the message bellow we may want to put the list of datasets too
+                    msg = "Making transfer subscriptions for {}".format(workflow['RequestName'])
+                    self.logger.info(msg)
+                    ddmResultList = self.ddm.makeAggRequests(ddmReqList, aggKey='item')
```

Comment:
Imagine that this `makeAggRequests` actually makes 3 calls to DDM. Where 1 fails and the other 2 return 200 response. How do you deal with the request that failed here? From L322, I'd say the code won't even noticed it, provided at least 1 request went through.

Reply:
Yes, I know, but could not find a way out of this that satisfies me enough. The alternative I was considering was having the transfer status as a list of dictionaries like the following: [1]. This will generally lead to information repetition, but will help if we need to do a reverse map of transfer Id to reqest status per workflow, and eventually in the next step when we start aggregate for many workflows..... but there (once we cross the boundaries of the workflow) things get messy anyway. What do you think?

[1]
```
transferstatus: [{'transferId': transferId1,
                  'status': status1},
                {'transferId': transferId2,
                  'status': status2},
...] 
```